EngineLayer/ModernSearch/ModernSearchEngine.cs

```diff
@@ -85,28 +85,42 @@ protected override MetaMorpheusEngineResults RunSpecific()
                     for (int possibleWinningPeptideIndex = 0; possibleWinningPeptideIndex < fullPeptideScores.Length; possibleWinningPeptideIndex++)
                     {
                         var consideredScore = fullPeptideScores[possibleWinningPeptideIndex];
-                        CompactPeptide candidatePeptide = peptideIndex[possibleWinningPeptideIndex];
-                        for (int j = 0; j < searchModesCount; j++)
+                        if (consideredScore > scoreCutoff) //intentionally high. 99.9% of 4-mers are present in a given UniProt database. This saves considerable time
                         {
-                            // Check if makes sense to add due to peptidescore!
-                            var searchMode = searchModes[j];
-                            double currentBestScore = bestScores[j];
-                            if (currentBestScore > 1)
+                            CompactPeptide candidatePeptide = peptideIndex[possibleWinningPeptideIndex];
+                            for (int j = 0; j < searchModesCount; j++)
                             {
-                                // Existed! Need to compare with old match
-                                if (Math.Abs(currentBestScore - consideredScore) < 1e-9)
+                                // Check if makes sense to add due to peptidescore!
+                                var searchMode = searchModes[j];
+                                double currentBestScore = bestScores[j];
+                                if (currentBestScore > 1)
```

Comment:
Not sure that matters. That line is just to determine if a currentBestScore exists. We could set it to scoreCutoff, but it will accomplish the same thing...